src/main/java/uk/gov/laa/ccms/caab/mapper/AssessmentMapper.java

```diff
@@ -0,0 +1,497 @@
+          .entities(globalEntityList);
+    }
+  }
+
+  /**
+   * Converts the proceeding context within the given assessment mapping context to an
+   * {@link AssessmentEntityTypeDetail}.
+   *
+   * @param context the assessment mapping context containing proceeding details
+   * @return the assessment entity type detail for proceedings
+   */
+  default AssessmentEntityTypeDetail toAssessmentEntityTypeDetailProceeding(
+      final AssessmentMappingContext context) {
+
+    if (context == null) {
+      return null;
+    }
+
+    final List<ProceedingDetail> proceedings = context.getApplication().getProceedings();
+
+    final AssessmentEntityTypeDetail existingEntityType =
+        getAssessmentEntityType(context.getAssessment(), PROCEEDING);
+
+    if (existingEntityType != null) {
+      return existingEntityType;
+    } else {
+      final List<AssessmentEntityDetail> proceedingEntityList = new ArrayList<>();
+
+      if (proceedings != null && !proceedings.isEmpty()) {
+        proceedingEntityList.addAll(
+            toAssessmentEntityDetailListProceeding(proceedings));
+      }
+
+      return new AssessmentEntityTypeDetail()
+          .name(PROCEEDING.getType())
+          .entities(proceedingEntityList);
+    }
+  }
+
+  /**
+   * Converts the opponent context within the given assessment mapping context to an
+   * {@link AssessmentEntityTypeDetail}.
+   *
+   * @param context the assessment mapping context containing opponent contexts
+   * @return the assessment entity type detail for opponents
+   */
+  default AssessmentEntityTypeDetail toAssessmentEntityTypeDetailOpponent(
+      final AssessmentMappingContext context) {
+
+    if (context == null) {
+      return null;
+    }
+
+    final List<AssessmentOpponentMappingContext> opponentContextList =
+        context.getOpponentContext();
+
+    final AssessmentEntityTypeDetail existingEntityType =
+        getAssessmentEntityType(context.getAssessment(), OPPONENT);
+
+    if (existingEntityType != null) {
+      return existingEntityType;
+    } else {
+      final List<AssessmentEntityDetail> opponentsEntityList = new ArrayList<>();
+
+      if (opponentContextList != null && !opponentContextList.isEmpty()) {
+        opponentsEntityList.addAll(
+            toAssessmentEntityDetailListOpponent(opponentContextList));
+      }
+
+      return new AssessmentEntityTypeDetail()
+          .name(OPPONENT.getType())
+          .entities(opponentsEntityList);
+    }
+  }
+
+  List<AssessmentEntityDetail> toAssessmentEntityDetailListProceeding(
+      final List<ProceedingDetail> proceedings);
+
+
+  /**
+   * Converts proceeding data to a list of {@link AssessmentAttributeDetail} objects.
+   *
+   * @param proceeding the proceeding detail containing various attributes
+   * @return a list of assessment attribute details for the proceeding
+   */
+  @Named("proceedingToAttributeList")
+  default List<AssessmentAttributeDetail> proceedingToAttributeList(
```

Comment:
Please can these be compared to the files:

com.legalservices.laa.ccms.config.MappingConfig In PUI

see - public CcmsInstancePrepopulator foreachProceeding() {